File: proj/Logic/src/quoridor/logic/Move.java
package quoridor.logic;

public class Move {
	
	public static int MOVE_PAWN = 0;
	public static int SET_WALL = 1;
	
	public int type;
	public int row;
	public int col;
	public boolean horizontal;

	public Move(int pRow, int pCol) {
		type = MOVE_PAWN;
		row = pRow;
		col = pCol;
	}
	
	public Move(int pRow, int pCol, boolean pHorizontal) {
		type = SET_WALL;
		row = pRow;
		col = pCol;
		horizontal = pHorizontal;
	}
	
	public String toString() {
		if(type == MOVE_PAWN) {
			return "Move pawn by (" + row + "," + col + ")";
		}
		if(type == SET_WALL) {
			return "Set wall ate (" + row + "," + col + ") [" + horizontal + "]";
		}
		
		return "INVALID";
	}
}

File: Assets/Scripts/GameAI/Move.cs
using UnityEngine;
using System.Collections;

public class Move
{
    public const int MovePawn = 0;
    public const int SetWall = 1;

    public int Type;
    public int Row, Col;
    public bool IsHorizontal;

    public Move(Move move)
    {
        Type = move.Type;
        Row = move.Row;
        Col = move.Col;
        IsHorizontal = move.IsHorizontal;
    }

    public Move(int row, int col)
    {
        Type = MovePawn;
		Row = row;
		Col = col;
    }

    public Move(int row, int col, bool horizontal)
    {
        Type = SetWall;
        Row = row;
        Col = col;
        IsHorizontal = horizontal;
    }

    public override string ToString()
    {
        if (Type == MovePawn)
        {
            return "Move Pawn to: " + Row + " " + Col;
        }
        else
        {
            return "Set Wall at: " + Row + " " + Col + " (" + IsHorizontal + ")";
        }
    }
}
